File: lista3/zadanie2/src/Hexagon.cpp
#include <cmath>

#include "Figure.hpp"
#include "ClientInput.hpp"
#include "Hexagon.hpp"

Hexagon::Hexagon(ClientInput input) : sideLength(input.NextInput())
{}

double Hexagon::Area()
{
    return 3. * pow(sideLength, 2.) * sqrt(3.) / 2.;
}

double Hexagon::Perimeter()
{
    return 6. * sideLength;
}

std::string Hexagon::NameYourself()
{
    return "Szesciań";
}

File: lista3/zadanie1/Hexagon.java
public class Hexagon extends Figure
{
    public Hexagon(ClientInput input)
    {
        sideLength = input.NextInput();
    }

    public double Area()
    {
        return 3. * Math.pow(sideLength, 2.) * Math.sqrt(3.) / 2.;
    }

    public double Perimeter()
    {
        return 6. * sideLength;
    }

    public String NameYourself()
    {
        return "Szesciań";
    }

    private double sideLength;
};
